fillers/eips/eip4844/datahash_opcode.py

```diff
@@ -538,118 +545,238 @@ def test_datahash_gas(_: str):
         timestamp=1000,
     )
 
-    datahash_zero_gas_measure_code = CodeGasMeasure(
-        code=(Op.PUSH1(0x00) + Op.DATAHASH),
-        overhead_cost=3,
-        extra_stack_items=1,
-    )
-    datahash_zero_gas_measure_code_address = to_address(0x100)
-
-    datahash_max_gas_measure_code = CodeGasMeasure(
-        code=(Op.PUSH32(2**256 - 1) + Op.DATAHASH),
-        overhead_cost=3,
-        extra_stack_items=1,
+    # Initial measures with zero, max and random
+    gas_measures = [0x00, 2**256 - 1]
+    gas_measures.extend(
+        [
+            random.randint(0, 2**256 - 1)
+            for _ in range(MAX_BLOB_PER_BLOCK * 10)
+        ]
     )
-    datahash_max_gas_measure_code_address = to_address(0x200)
+    gas_measures_code = [
+        CodeGasMeasure(
+            code=getattr(Op, "PUSH32")(gas_measure) + Op.DATAHASH,
+            overhead_cost=3,
+            extra_stack_items=1,
+        )
+        for gas_measure in gas_measures
+    ]
 
-    # Store all contracts in pre-state
     pre = {
         TestAddress: Account(balance=1000000000000000000000),
-        datahash_zero_gas_measure_code_address: Account(
-            code=datahash_zero_gas_measure_code
-        ),
-        datahash_max_gas_measure_code_address: Account(
-            code=datahash_max_gas_measure_code
-        ),
     }
+    post = {}
+
+    def create_tx(ty: int, to: str, nonce: int):
+        return Transaction(
+            ty=ty,
+            nonce=nonce,
+            data=to_hash_bytes(0),
+            to=to,
+            gas_limit=3000000,
+            max_fee_per_gas=10,
+            max_priority_fee_per_gas=10,
+            max_fee_per_data_gas=10,
```

Comment:
I'm not sure if this could cause issues (`max_fee_per_data_gas != None`) for type < 5.

We could set it to `None` for those cases just to be safe.